cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -34,6 +35,18 @@
  */
 public class QueryResult implements Iterable<DocumentRevision> {
 
+    /**
+     * This unchecked exception is used to wrap another exception so
+     * we can pass meaningful exception traces over the {@link #iterator()}
+     * boundary as we can't change the signature of {@link #iterator} to
+     * add a checked exception.
+     */
+    public class UncheckedDocumentException extends RuntimeException {
+        public UncheckedDocumentException(Exception causedBy){
+            super(causedBy);
+        }
+    }
```

Comment:
Making this exception checked would require alot of work due to the `Iterable` interface, so I think we should raise a ticket for that, and maybe try and get to it before 1.0